File: builder/SlackNotification.java
package builder;

public class SlackNotification {

    private String content;
    private String sender;
    private String recipient;
    private String timestamp;
    private boolean hasMarkdownLanguage;

    SlackNotification(SlackNotificationBuilder slackNotificationBuilder) {
        this.content = slackNotificationBuilder.getContent();
        this.sender = slackNotificationBuilder.getSender();
        this.recipient = slackNotificationBuilder.getRecipient();
        this.timestamp = slackNotificationBuilder.getTimestamp();
        this.hasMarkdownLanguage = slackNotificationBuilder.isHasMarkdownLanguage();
    }

    public String getContent() {
        return content;
    }

    public String getSender() {
        return sender;
    }

    public String getRecipient() {
        return recipient;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public boolean isHasMarkdownLanguage() {
        return hasMarkdownLanguage;
    }
}

File: C#/Builder/SlackNotification.cs
using System.Text.Json.Serialization;

namespace Builder;

public sealed class SlackNotification
{
    internal SlackNotification(SlackNotificationBuilder slackNotificationBuilder)
    {
        Content = slackNotificationBuilder.Content;
        Sender = slackNotificationBuilder.Sender;
        Recipient = slackNotificationBuilder.Recipient;
        Timestamp = slackNotificationBuilder.Timestamp;
        HasMarkdownLanguage = slackNotificationBuilder.HasMarkdownLanguage;
    }

    public string? Content { get; }
    public string? Sender { get; }
    public string? Recipient { get; }
    public string? Timestamp { get; }
    public bool HasMarkdownLanguage { get; }
}

[JsonSerializable(typeof(SlackNotification))]
public sealed partial class SlackNotificationJsonContext : JsonSerializerContext;